### ISSUE TITLE
Emote menu is almost empty

### ISSUE THREAD

**Issue by Honkizz** [night/BetterTTV#2787]
The emote menu has only 3 emotes ClappyHype, ClappyDerp and Clappy.

**Comment by night** [comment]
Not able to reproduce this issue. Try disabling third party extensions and make sure you’re logged into Twitch

**Comment by xgerhard** [comment]
I do see this happening sometimes, sometimes a hard refresh fixes it.
It loads the BTTV channel emotes, then the new clips emotes as Honkizz describes, the rest fails/is empty.
No errors in console, according to the network tab, all emotes are loaded in, but for some reason not added to the menu.

To reproduce this seems to work for me on Chrome (With only the BTTV extension enabled):
- Open incognito tab
- Go to a random channel
- Sign in with Twitch
- Enable BTTV emote menu
- Close settings
- Open BTTV emote menu

Hopefully this helps to resolve it.

**Comment by BradyThe** [comment]
I had the same problem with my default browser. The emote menu looked like this:
![firefox_2017-11-21_22-37-22](https://user-images.githubusercontent.com/8181555/33099248-863db298-cf10-11e7-8859-6c71b3d7624b.png)

I precautionally cleared all cookies from Firefox, and to eliminate any incompatibilities or override I tested it on different browsers:

Edge (never used before)
![applicationframehost_2017-11-21_22-54-23](https://user-images.githubusercontent.com/8181555/33099529-76a8b3b8-cf11-11e7-8536-6acb2ebcd5ee.png)

Chrome (used before, different profile)
![chrome_2017-11-21_22-55-16](https://user-images.githubusercontent.com/8181555/33099547-86f69f82-cf11-11e7-85bd-e03a52d3523c.png)

**Comment by BradyThe** [comment]
I had the same problem with my default browser (Firefox). The emote menu looked like this:
![firefox_2017-11-21_22-37-22](https://user-images.githubusercontent.com/8181555/33099571-98faf3d6-cf11-11e7-85fe-7a4b780846f6.png)

I precautionally cleared all cookies from Firefox, and to eliminate any incompatibilities or override I tested it on different browsers with a Twitch account that makes use of BetterTTV channel emotes:

Edge (never used before)
![applicationframehost_2017-11-21_22-54-23](https://user-images.githubusercontent.com/8181555/33099783-661b75fc-cf12-11e7-8642-e2903b22fe3b.png)

Chrome (used before, different profile)
![chrome_2017-11-21_22-55-16](https://user-images.githubusercontent.com/8181555/33099587-a2347418-cf11-11e7-9cf5-56e07c1da11f.png)

Firefox (initially the emote menu did not work even after disabling/clearing/refreshing, but somehow it started working globally after visiting this channel, is probably just a coincidence)
![firefox_2017-11-21_22-58-21](https://user-images.githubusercontent.com/8181555/33099644-d50cf216-cf11-11e7-994c-aca866d7a3b8.png)

Note: My emote menu might look different because I hide most emotes.

**Comment by xepheratu** [comment]
I have found in my trial and error phase of checking things, that third party cookies is a big factor in how this works, if third party cookies are enabled, then all of the emotes show up, but if third party cookies are disabled then this issue happens.

**Comment by NoTraceFound** [comment]
I just tested unblocking cookies from "cdn.krxd.net" in my privacybadger addon and that fixed the issue, seems to be a CDN needed to get some info for some reason?

**Comment by night** [comment]
It’s not a cookies issue. It’s most likely a race condition. Subsequent loads have populated caches, which is why refreshing fixes it.

**Comment by NoTraceFound** [comment]
Yeah I retract my statement, refreshing seems to have a completely random chance of making it work or breaking it again, huh

**Comment by u-C-m** [comment]
Not sure if this belongs here, but I got this bug alongside with the "empty" emote menu. Could be a coincidence tho.
From time to time my twitch chat shows some info for the bttv emotes and it wont go away. Other emotes are almost completete cut off on the left side of the chat.

![twitch_chat](https://user-images.githubusercontent.com/28190506/33692782-63dc2780-daef-11e7-9026-96bc6b8bd95d.png)

**Comment by Lonewolf22113** [comment]
My emote menu is completely empty no matter how many times i refresh :(

![image](https://i.imgur.com/p7DOHzA.png)

**Comment by DAOWAce** [comment]
Emote menu doesn't seem to be working whatsoever with the new update.

**Comment by prizna** [comment]
Happening for me too, emote menu is mostly empty, only showing 1 lot of subscriber emotes (im subscribed to more) a few bttv channel emotes and nothing else. Refeshing sometimes works but not always.

**Comment by xepheratu** [comment]
After much trial and error, blocking and allowing addons and domains, I still find it very random, and there seems to be no rhyme nor reason to it sometimes. I use the addons decentraleyes, cookie autodelete, privacybadger, ublock origin, and https everywhere. I have disabled all of the addons and it will still sometimes not work. I have disabled only decentraleyes and it will work, but upon a refresh of the page, they disappear again, again adding to my idea that it is somewhat random for some reason. Upon disabling privacybadger, they do not show up again. Disabling https everywhere makes them stay hidden as well. After disabling ublock origin, they do not show up again. Meaning the only addon left is cookie autodelete, I disable it, and it simply does not show up again. I have disabled every addon aside from bttv and restarted firefox, and they still dont show up. So I believe it is either a problem in the code, or a server side something which is not allowing them all to load because as it isnt consistent, it must be something of a server simply not loading or accessing properly, which even having no addons does not fix.

I just thought I would add my info and try to help.

**Comment by Doorstop2** [comment]
Been having this problem also sometimes it loads sometimes it doesn't but the channel BTTV emotes always load the global ones not so much.

**Comment by Notjubby** [comment]
Same issue for a while now in chrome. All I see the is "Favorite Emotes" section and nothing else, even the section titles are empty.

**Comment by trigunflame** [comment]
Any chance of fixing this? 

Twitch life FeelsBadMan for the past several days/weeks this shit hasn't been working.

**Comment by valdor27** [comment]
I have the same problem for a few days.

**Comment by DayBr3ak** [comment]
This solves the problem for me. I've posted that on Aaron128 fork, but it's basically removing one line.

https://github.com/Aaron128l/Userscript--Twitch-Chat-Emotes/pull/1

**Comment by GraveGalius** [comment]
I just tried to mess around with third party extentions that i had while hard refreshing the page (CTRL + F5) and this fixed the problem for me, first it worked disabling "4chan image expander and saver" and AdBlockPlus, but when i re-enabled ABP the emote menu was empty again, so i tried to leave ABP on and disable other to test if it was ABP the problem. All turned fine again when i disabled "Simple Image Hover Zoom" and it isn't giving more problems for now.
TL;DR I think that every extention related to image manipulation (zoom, hover zoom, phisically moving images) creates conflicts betwenn ABP and the emote menu.

btw don't worry, i keep ABP disabled for all the twitch site ;)
